datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/config.py

```diff
@@ -18,73 +21,88 @@
+            continue
 
-                # We must convert to text here to free Exception object before it goes out of scope
-                error = str(e)
-
-                check_display_queue.append(lambda: echo_info('{}:'.format(file_name), indent=True))
-                check_display_queue.append(lambda: echo_failure('Invalid YAML -', indent=FILE_INDENT))
-                check_display_queue.append(lambda: echo_info(error, indent=FILE_INDENT * 2))
-                continue
-
-            errors = validate_config(file_data)
-            for err in errors:
-                err_msg = str(err)
-                if err.severity == SEVERITY_ERROR:
-                    file_display_queue.append(lambda x=err_msg: echo_failure(x, indent=FILE_INDENT))
-                    files_failed[config_file] = True
-                elif err.severity == SEVERITY_WARNING:
-                    file_display_queue.append(lambda x=err_msg: echo_warning(x, indent=FILE_INDENT))
-                    files_warned[config_file] = True
-                else:
-                    file_display_queue.append(lambda x=err_msg: echo_info(x, indent=FILE_INDENT))
+        # Just use six to make it easier to search for occurrences of text we need to remove when we drop Python 2
+        if PY2:
+            check_display_queue.append(
+                lambda **kwargs: echo_failure('Dictionary key order is only guaranteed in Python 3.7.0+', **kwargs)
+            )
 
-            # Verify there is an `instances` section
-            if 'instances' not in config_data:
-                files_failed[config_file] = True
-                file_display_queue.append(lambda: echo_failure('Missing `instances` section', indent=FILE_INDENT))
+        file_counter.append(None)
 
-            # Verify there is a default instance
-            else:
-                instances = config_data['instances']
-                if check not in IGNORE_DEFAULT_INSTANCE and not isinstance(instances, list):
-                    files_failed[config_file] = True
-                    file_display_queue.append(lambda: echo_failure('No default instance', indent=FILE_INDENT))
+        # source is the default file name
+        if repo_choice == 'agent':
+            display_name = 'Datadog Agent'
+            source = 'datadog'
+        else:
+            display_name = load_manifest(check).get('display_name', check)
+            source = check
 
-            if file_display_queue:
-                check_display_queue.append(lambda x=file_name: echo_info('{}:'.format(x), indent=True))
-                check_display_queue.extend(file_display_queue)
+        spec = ConfigSpec(read_file(spec_path), source)
+        spec.load()
+
+        if spec.errors:
+            files_failed[spec_path] = True
+            for error in spec.errors:
+                check_display_queue.append(lambda error=error, **kwargs: echo_failure(error, **kwargs))
+        else:
+            if spec.data['name'] != display_name:
```

Comment:
It should be what users see, basically the name of the product. As far as stability, isn't `display_name` the one that cannot/should not be changed (backend-wise)?

Reply:
If we want to display a name, should we just retrieve it from `manifest.json` instead of duplicating that information?

> As far as stability, isn't display_name the one that cannot/should not be changed (backend-wise)?

AFAIK we try to make it changeable in the future.

Reply:
Yeah, currently the `display_name` of an integration can't change once its merged in. `public_title` is the field that can safely change.